HLTrigger/HLTfilters/plugins/L1TTkMuonFilter.cc

```diff
@@ -145,23 +147,31 @@ L1TTkMuonFilter::MuonQualityCut::MuonQualityCut(const edm::ParameterSet& iConfig
   }
 }
 
-bool L1TTkMuonFilter::MuonQualityCut::operator()(const l1t::TkMuon& muon) const {
-  const auto& qualities = allowedQualities_.find(muon.muonDetector());
-  if (qualities != allowedQualities_.end()) {
-    return std::binary_search(qualities->second.begin(), qualities->second.end(), muon.quality());
-  } else {
-    return true;  //if qualities for that detector is not specified, we return true
+bool L1TTkMuonFilter::MuonQualityCut::operator()(const l1t::TrackerMuon& muon) const {
+  // If we didn't load any qualities from the config file, that means we don't care.
+  // So we set passesQuality to true.
+  if (allowedQualities_.empty()) {
+    return true;
   }
+
+  bool passesQuality(false);
+  // The muonDetector() method is not available for TrackerMuon (it was available in TkMuon),
+  // so for the time being we just take the union of all quality vectors for all muon detectors.
```

Comment:
I agree this is a pain, but if the `TrackerMuon` do not have the possibility to distinguish the different muon detectors, it would be better to change the interface of the module (_i.e._ the configuration parameters) to take a single list of qualities, instead of the more complex approach there is now.

Reply:
I was under the impression that it would be better to **not** change the interface of the module, but I am okay doing the change anyway.

Reply:
hi @trtomei, here is my point of view:
  - if the module interface (its configuration parameters) advertises a feature that is never going to be available, we should update the interface to match what is available; keeping the old parameters week only create confusion down the road;
  - if the old behaviour can be recovered (for example, if L1 will again provide the information about which track finder built each candidate) or emulated (for example, based on hardcoded eta ranges), it would be OK to keep the current configuration for the time being; I would suggest adding a comment about this in the code and in the python configuration inside the HLT menu;
  - if the old behaviour cannot be recovered, but it's useful and could be emulated with a different interface (for example, based on eta ranges defined in the confusion), I would suggest making that change, or at least leaving a comment for the time being.